Formula/dlib.rb

```diff
@@ -28,6 +32,16 @@ def install
     args = std_cmake_args + %w[-DDLIB_USE_BLAS=ON -DDLIB_USE_LAPACK=ON]
     args << "-DDLIB_NO_GUI_SUPPORT=ON" if build.without? "x11"
 
+    if build.with? "avx"
+      args << "-DUSE_AVX_INSTRUCTIONS=ON"
+    end
+    if build.with? "sse4"
+      args << "-DUSE_SSE4_INSTRUCTIONS=ON"
+    end
+    if build.with? "sse2"
+      args << "-DUSE_SSE2_INSTRUCTIONS=ON"
```

Comment:
SSE2 can be defined unconditionally as it is present on all modern macOS hardware.